flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -115,8 +115,12 @@ public boolean isRegisteredAsAvailable() {
 	@Override
 	public boolean isAvailable() {
 		// BEWARE: this must be in sync with #isAvailable(BufferAndBacklog)!
-		return hasBuffersAvailable() &&
-			(numCreditsAvailable > 0 || subpartitionView.nextBufferIsEvent());
+		return hasBuffersAvailable() && !isBlocked();
+	}
+
+	@Override
+	public boolean isBlocked() {
```

Comment:
I think the `isBlocked` naming can not indicate the specific semantics. How about changing to `isCreditsAvailable()` directly?

To do so, the `isAvailable()` breaks into `buffersAvailable()` and `creditsAvailable()`, and the following private method `isAvailable(BufferAndBacklog bufferAndBacklog)` may also reuse `isCreditsAvailable()` .